transformation/src/main/scala/org/broadinstitute/monster/hca/HcaLog.scala

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.monster.hca
+
+import org.broadinstitute.monster.hca.PostProcess.errorCount
+import org.slf4j.Logger
+import ujson.Obj
+
+// a simple log level case class hierarchy to make sure we log at the right level
+trait LogLevel
+case class HcaErrorLog() extends LogLevel
+case class HcaWarnLog() extends LogLevel
+
+// a trait to capture the generic logging mechanism we'll want, with case classes for the different logging levels
+abstract class HcaLog {
+  val level: LogLevel
+  val jsonMsg: Obj
+
+  def log(implicit logger: Logger): Unit =
+    level match {
```

Comment:
I don't think `level` is necessary. If the `log` method is `abstract` here then you can override it in `HcaErrorLog` and `HcaWarnLog` to do the correct thing for each case.

Reply:
Ah good call, I like that approach better.